### ISSUE TITLE
docs: provide autocomplete documentation for "magical" fields

### ISSUE THREAD

**Issue by Hotell** [nrwl/nx-console#1385]
## Description

Only `project` schema property name triggers autocomplete for all workspace projects within nx-console.

## Motivation

Our custom generators are using `name` property in generator schemas. That `name` is supposed to be existing project name in workspace.

It would be great to have official documentation about these magic behaviours, and ideally make them configurable. 

While we can refactor `name` to `project` it would introduce breaking change and some amount of confusion to our users with the sudden change.

## Suggested Implementation

- docs for magic fields in generators/executors schemas
- configurability of magic(special) schema fields

**Comment by Hotell** [comment]
So it's kinda documented within nx source code :D 

this does it
```diff
"properties": {
    "name": {
      "type": "string",
      "description": "Library name or comma delimited library names to execute migration on multiple libraries.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
+      "x-dropdown": "projects",
      "pattern": "^[@a-zA-Z].*$"
    },
```

⬇️⬇️⬇️

<img width="378" alt="image" src="https://user-images.githubusercontent.com/1223799/196506090-7fabfb81-5850-4108-9c7c-4e06cc0a10ef.png">

**Comment by Cammisuli** [comment]
You're right 🙂 it's on the next thing I'm planning on doing for Nx Console. Get better documentation out.
